test/integration/apiserver/apply/apply_crd_test.go

```diff
@@ -733,3 +735,206 @@ spec:
 		t.Fatalf("Expecting to get one conflict when a different applier updates existing list item, got: %v", status.Status().Details.Causes)
 	}
 }
+
+// This tests for cases when conversion logic is called on managed field objects
+// in a CR: if a CR exists with a managedFields entry whose API Version is a version
+// that no longer exists, if a SSA operation is now done on this CR, the API Server
+// tries to perform conversion and fails. This should *not* hinder workflows and
+// updates should still be possible.
```

Comment:
So the logic of SSA here is that it will remove the entry of the wrong version specifically, but not the others. So I would suggest that maybe the test includes:
- Have entries in old and new versino
- Remove old version
- Make sure entries initially written in new version persist (iow, make sure we haven't deleted all entries)

Reply:
I'm not sure I understand
> make sure we haven't deleted all entries

Do you mean in the end we check that `managedFields` is not empty?

Reply:
I don't know if not empty is strong enough, but we need to make sure this doesn't delete entries that shouldn't be deleted.